src/Init/Data/String/Basic.lean

```diff
@@ -89,13 +123,38 @@ def utf8SetAux (c' : Char) : List Char → Pos → Pos → List Char
   | c::cs, i, p =>
     if i = p then (c'::cs) else c::(utf8SetAux c' cs (i + c) p)
 
+/--
+Replaces the character at a specified position in a string with a new character.
+
+If the position is invalid, returns the string unchanged.
+* `"abc".set ⟨1⟩ 'B' = "aBc"`
+* `"abc".set ⟨3⟩ 'D' = "abc"`
+* `"L∃∀N".set ⟨1⟩ 'X' = "L∃∀N" -- invalid position as '∃' is a mutli-byte character`
+
+If both the replacement character and the replaced character are ASCII characters and the string is not shared, destructive updates are used.
+-/
 @[extern "lean_string_utf8_set"]
 def set : String → (@& Pos) → Char → String
   | ⟨s⟩, i, c => ⟨utf8SetAux c s 0 i⟩
 
+/--
+Applies `f` to the character at position `p` in the string `s`.
+
+If `p` is an invalid position, returns the string unchanged.
+* `abc.modify ⟨1⟩ Char.toUpper = "aBc"`
+* `abc.modify ⟨3⟩ Char.toUpper = "abc"`
+-/
 def modify (s : String) (i : Pos) (f : Char → Char) : String :=
   s.set i <| f <| s.get i
 
+/--
+Returns the next position in a string after `p` given that `p` is a valid position.
```

Comment:
@nomeata  I updated the docstring for `next` after considering the multi-byte char cases more carefully. It seems to me that `p` being a valid position (lying on a UTF-8 boundary) is an expectation of this function. 

In the case where `p` doesn't lie on a UTF-8 byte boundary. For instance, `"L∃∀N".next ⟨2⟩ = 3` since `"L∃∀N".get ⟨2⟩ = (default : Char) = 'A'`.

There is also the separate `next'` which requires `(h : ¬ s.atEnd p)` but it doesn't include the multi-byte char case.

What do you think?

Reply:
Much better wording. Thanks!

I also added the case where `p = s.endPos` as it also results in an invalid position.